training/src/app/modules/chuck-norris/components/search/search.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, Output, EventEmitter, OnInit } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { debounceTime, Subject, takeUntil, tap } from "rxjs";
+
+@Component({
+	selector: 'app-search',
+	templateUrl: './search.component.html',
+	styleUrls: ['./search.component.scss'],
+
+})
+export class SearchComponent implements OnInit {
```

Comment:
i think the solution for this component is:   1. event that fire the search string when typing.   2. event that fire after the debounce time.  if this is what u tried to do please rename events and methods, its not clear